rclcpp/src/rclcpp/executor.cpp

```diff
@@ -42,28 +43,35 @@ using rclcpp::FutureReturnCode;
 
 Executor::Executor(const rclcpp::ExecutorOptions & options)
 : spinning(false),
+  shutdown_guard_condition_(std::make_shared<rclcpp::GuardCondition>(options.context)),
   memory_strategy_(options.memory_strategy)
 {
+  // Store the context for later use.
+  context_ = options.context;
+
   rcl_guard_condition_options_t guard_condition_options = rcl_guard_condition_get_default_options();
   rcl_ret_t ret = rcl_guard_condition_init(
-    &interrupt_guard_condition_, options.context->get_rcl_context().get(), guard_condition_options);
+    &interrupt_guard_condition_, context_->get_rcl_context().get(), guard_condition_options);
   if (RCL_RET_OK != ret) {
     throw_from_rcl_error(ret, "Failed to create interrupt guard condition in Executor constructor");
```

Comment:
This line isn't hit anymore.
The reason is that now we're creating two guard conditions, instead of one, and we're hitting the error in the first creation and not here.

Reply:
@brawner Any suggestion to cover this line without adding a mock that fails in the second call to `rcl_guard_condition_init`?
I can apply the mentioned option, but IMO that's a bit "strange".

Maybe fault injection works for this, I will try to give it a try.

P.S.: I don't think that hitting this line matters, as overall coverage has improved.

Reply:
> P.S.: I don't think that hitting this line matters, as overall coverage has improved.

Yeah, that's my opinion too.  If overall coverage is improving, then its OK to lose a line like this.  If it becomes important, we can always revisit it later.

Reply:
The fault injection tools were intended for situations like this, but I think it's fine without.